airflow/config_templates/default_airflow.cfg

```diff
@@ -565,6 +565,9 @@ dag_dir_list_interval = 300
 # How often should stats be printed to the logs
 print_stats_interval = 30
 
+# A DagRun can have a 'removed' task instance and still lead to DagRun 'success' state when False
```

Comment:
Reading this I don't know what this does.

Can you explain (to me here) what this does, so we can come up with a better doc comment here.

Reply:
Right now when evaluating the overall state of a DagRun the overall DagRun state can be set to SUCCESS even though one of the TaskInstance is in REMOVED state. With this new config set to True, when evaluating the overall state of a DagRun the overall DagRun state will be set to FAILED if any TaskInstance is in REMOVED state